auth-tokens/src/main/java/com/palantir/tokens/auth/BearerToken.java

```diff
@@ -32,6 +40,47 @@
     private static final String VALIDATION_PATTERN_STRING = "^[A-Za-z0-9\\-\\._~\\+/]+=*$";
     private static final Pattern VALIDATION_PATTERN = Pattern.compile(VALIDATION_PATTERN_STRING);
 
+    private static final String SUBJECT_FIELD = "sub";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * If the token contains a 'sub' field, as a base64 encoded UUID, return it as a String.
+     * @return the 'sub' field of the JWT, if it exists
+     */
+    public final Optional<String> getUserIdInsecure() {
```

Comment:
The ones that are used by Palantir are, right? I don't think we need to compromise for non-Palantir use cases.

Beyond that, I think that this is fine - if it's not a JWT in the format generated by our internal auth solution, you just don't get a user id back.